src/system/SystemPacketBuffer.h

```diff
@@ -257,12 +338,45 @@ class DLL_EXPORT PacketBuffer : private pbuf
      */
     CHECK_RETURN_VALUE PacketBufferHandle Last();
 
+    /**
+     * Perform an implementation-defined check on the validity of a PacketBuffer pointer.
+     *
+     * Unless enabled by #CHIP_CONFIG_MEMORY_DEBUG_CHECKS == 1, this function does nothing.
+     *
+     * When enabled, it performs and implementation- and configuration-defined check on
+     * the validity of the packet buffer. It MAY log an error and/or abort the program
+     * if the packet buffer or the implementation-defined memory management system is in
+     * a faulty state. (Some configurations may not actually perform any check.)
+     *
+     * @note  A null pointer is not considered faulty.
+     *
+     *  @param[in] buffer - the packet buffer to check.
+     */
+    static void Check(const PacketBuffer * buffer)
+    {
+#if CHIP_SYSTEM_PACKETBUFFER_HAS_CHECK
+        InternalCheck(buffer);
+#endif
+    }
+
 private:
-#if !CHIP_SYSTEM_CONFIG_USE_LWIP && CHIP_SYSTEM_CONFIG_PACKETBUFFER_MAXALLOC
-    static PacketBuffer * sFreeList;
+    // Memory required for a maximum-size PacketBuffer.
+    static constexpr uint16_t kBlockSize = PacketBuffer::kStructureSize + PacketBuffer::kMaxSizeWithoutReserve;
 
+#if CHIP_SYSTEM_PACKETBUFFER_STORE == CHIP_SYSTEM_PACKETBUFFER_STORE_CHIP_POOL || defined(DOXYGEN)
+    typedef union
+    {
+        pbuf Header;
+        uint8_t Block[PacketBuffer::kBlockSize];
+    } BufferPoolElement;
+    static BufferPoolElement sBufferPool[CHIP_SYSTEM_CONFIG_PACKETBUFFER_POOL_SIZE];
```

Comment:
Doxygen throws an error otherwise. I think doxygen is mistaken, because `sBufferPool` only ever appears inside the same `#if` condition, but I don't see another way to shut it up.

Reply:
OK.  Probably worth a comment explaining that, and that DOXYGEN is never defined when doing "real" builds, right?